Azure eventhubs bulk subscribe support (#3756)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md

      value: "myeventhubstoragecontainer"
```

## Sending and receiving multiple messages

Azure Eventhubs supports sending and receiving multiple messages in a single operation using the bulk pub/sub API.

### Configuring bulk publish

> To set the metadata for bulk publish operation, set the query parameters on the HTTP request or the gRPC metadata as documented [here]({{< ref pubsub_api >}}).

Reviewer: ```suggestion
To set the metadata for bulk publish operation, set the query parameters on the HTTP request or the gRPC metadata, [as documented in the API reference]({{< ref pubsub_api >}}).
```

| Metadata | Default |
|----------|---------|
| `metadata.maxBulkPubBytes` | `1000000` |

### Configuring bulk subscribe

> When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to [Subscribing messages in bulk]({{< ref "pubsub-bulk#subscribing-messages-in-bulk" >}}) for more details. Learn more about [the bulk subscribe API]({{< ref pubsub-bulk.md >}}).

Reviewer: ```suggestion
When subscribing to a topic, you can configure `bulkSubscribe` options. Refer to [Subscribing messages in bulk]({{< ref "pubsub-bulk#subscribing-messages-in-bulk" >}}) for more details and to learn more about [the bulk subscribe API]({{< ref pubsub-bulk.md >}}).
```

| Configuration | Default |
|---------------|---------|
| `maxMessagesCount` | `100` |
| `maxAwaitDurationMs` | `10000` |

> ## Configuring Checkpoint Frequency

Reviewer 2: I do not understand what this section is trying to cover. Can you explain its usage some more?
It really need an example here on what you are trying to cover in this feature. Can you provide a more complex scenario and example?

Author: Let me add more context and an example for the same. Thanks for the suggestion

Author: Done. Please review

> ## Configuring Checkpoint Frequency

Reviewer: ```suggestion
## Configuring checkpoint frequency
```

> When subscribing to a topic, you can configure the checkpointing frequency in a partition by setting the metadata in the HTTP or gRPC subscribe request as documented [here](https://docs.dapr.io/reference/api/pubsub_api/#http-request-2). It will enable checkpointing after the configured number of events within a partition event sequence. It can be disabled by setting the frequency to `0`.  Learn more about Checkpointing [here](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#checkpointing).

Reviewer 2: - If you keep the URL, than make this in [XXX]({{< ref "pubsub-api/#http-request-2" >}}) format
- Best to replace "will" with present tense.

> When subscribing to a topic, you can configure the checkpointing frequency in a partition by setting the metadata in the HTTP or gRPC subscribe request as documented [here](https://docs.dapr.io/reference/api/pubsub_api/#http-request-2). It will enable checkpointing after the configured number of events within a partition event sequence. It can be disabled by setting the frequency to `0`.  Learn more about Checkpointing [here](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#checkpointing).

Reviewer: ```suggestion
When subscribing to a topic, you can configure the checkpointing frequency in a partition by [setting the metadata in the HTTP or gRPC subscribe request ]({{< ref "pubsub_api.md#http-request-2" >}}). This metadata enables checkpointing after the configured number of events within a partition event sequence. Disable checkpointing by setting the frequency to `0`.  

[Learn more about checkpointing](https://learn.microsoft.com/azure/event-hubs/event-hubs-features#checkpointing).
```

> Note: In case of BulkSubscribe, the checkpointing will occur after the configured number of batches, instead of events. 

Reviewer: ```suggestion
{{% alert title="Note" color="primary" %}}
When subscribing to a topic using `BulkSubscribe`, you configure the checkpointing to occur after the specified number of _batches,_ instead of events. 
{{% /alert %}}
```

> 

Reviewer 2: What does this note mean? What is configured number of batches ?

Author: Added description of a batch.

| Metadata | Default |
| -------- | ------- |

> | `metadata.checkPointFrequencyPerPartition` | `1` |

Reviewer 2: depending on how this is set, is this only set on subscriptions? Going back to the above, an example would really help here.

Reviewer: I second this

Author: Done
